### PR TITLE
Queues

### PR DESCRIPTION
**Pull request by lpil**

### CODE REVIEW

File: concepts/queues/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "blurb": "Lists in Gleam are immutable sequences of values that can be accessed quickly from the front or the back.",
```
Comment (giacomocavalieri) [inline concepts/queues/.meta/config.json:2]:
```suggestion
  "blurb": "Queues in Gleam are immutable sequences of values that can be accessed quickly from the front or the back.",
```

File: concepts/queues/about.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# About
+
+The Gleam standard library implements a `Queue` type in the `gleam/queue` module. It is similar to the `List` type, but with a few key differences:
+
+- The `Queue` type doesn't have a literal syntax that can be used to construct queues or pattern match on them.
+- Elements can be type can be efficiently added and removed from both the front and back of the `Queue` type, while the `List` type can only efficiently add and remove elements from the front.
```
Comment (giacomocavalieri) [inline concepts/queues/about.md:6]:
```suggestion
- Elements can be efficiently added and removed from both the front and back of the `Queue` type, while the `List` type can only efficiently add and remove elements from the front.
```

File: concepts/queues/about.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+let empty = queue.new()
+let one_to_four = queue.from_list([1, 2, 3, 4])
+```
+
+Elements can be added to the queue using the `queue.push_front` and `queue.push_back` functions:
+
+```gleam
+let one_to_four = queue.from_list([1, 2, 3, 4])
+let one_to_five = queue.push_back(one_to_four, 5)
+let zero_to_five = queue.push_front(one_to_five, 0)
```
Comment (giacomocavalieri) [inline concepts/queues/about.md:22]: Maybe I'd also show each queue in a comment, not really sure though, the name may be enough:
```suggestion
let one_to_four = queue.from_list([1, 2, 3, 4])
let one_to_five = queue.push_back(one_to_four, 5)
// -> queue.from_list([1, 2, 3, 4, 5])
let zero_to_five = queue.push_front(one_to_five, 0)
// -> queue.from_list([0, 1, 2, 3, 4, 5])
```

File: concepts/queues/about.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+```gleam
+let one_to_four = queue.from_list([1, 2, 3, 4])
+
+queue.pop_back(one_to_four)
+// -> Ok(#(3, queue.from_list([1, 2, 3])))
+
+queue.pop_front(one_to_four)
+// -> Ok(#(1, queue.from_list([2, 3, 4])))
+```
```
Comment (giacomocavalieri) [inline concepts/queues/about.md:35]: I'd explain why it returns a `Result` showing that it is `Error(Nil)` if the queue is empty

File: exercises/concept/magician-in-training/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+```gleam
+insert_top(queue.from_list([5, 9, 7, 1]), 8)
+// -> queue.from_list([5, 9, 7, 1, 8])
+```
+
+## 2. Remove the top card from the stack
+
+Implement the function `remove_top_card` that returns a copy of the stack which has had the card at the top of the stack removed. If the given stack is empty, the original stack should be returned, unchanged.
+
+```swift
```
Comment (giacomocavalieri) [inline exercises/concept/magician-in-training/.docs/instructions.md:20]:
```suggestion
```gleam
```

File: exercises/concept/magician-in-training/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+remove_bottom_card(queue.from_list([8, 5, 9, 7, 1]))
+// -> queue.from_list([5, 9, 7, 1])
+```
+
+## 5. Check size of the stack
+
+Implement the function `check_size_of_stack` that checks whether the size of the stack is equal to a given `stack_size` or not.
+
+```swift
```
Comment (giacomocavalieri) [inline exercises/concept/magician-in-training/.docs/instructions.md:47]:
```suggestion
```gleam
```

**Comment by lpil** [comment]
Thanks!

**Pull request merged**